engines/director/debugger.cpp

```diff
@@ -34,24 +37,147 @@ Debugger *g_debugger;
+	debugPrintf(" vars - Lists all of the variables available in the current script frame\n");
+	debugPrintf(" step / s [n] - Steps forward one or more operations\n");
+	debugPrintf(" next / n [n] - Steps forward one or more operations, skips over calls\n");
+	debugPrintf(" finish / fin - Steps until the current stack frame returns/n");
+	debugPrintf("\n");
+	debugPrintf("Breakpoints:\n");
+	debugPrintf("\n");
+	//debugPrintf(" bpset [funcname:n] - Creates a breakpoint on a Lingo script/n");
+	//debugPrintf(" bpdel [n] - Deletes a specific breakpoint /n");
+	return true;
+}
+
+bool Debugger::cmdVersion(int argc, const char **argv) {
+	debugPrintf("Director version: %d\n", g_director->getVersion());
+	debugPrintf("Director platform: %s\n", Common::getPlatformCode(g_director->getPlatform()));
+	debugPrintf("Game ID: %s\n", g_director->getGameId());
+	debugPrintf("Game name: %s\n", getDescriptionFromGameId(g_director->getGameId()));
+	debugPrintf("Game variant: %s\n", g_director->getExtra());
+	debugPrintf("Language: %s\n", Common::getLanguageCode(g_director->getLanguage()));
+	debugPrintf("Expected Director version: %d\n", g_director->getDescriptionVersion());
+	debugPrintf("Executable name: %s\n", g_director->getEXEName().c_str());
+	debugPrintf("\n");
+	return true;
+}
+
+bool Debugger::cmdRepl(int argc, const char **argv) {
+	debugPrintf("Switching to Lingo REPL mode, type 'exit' to return to the debug console.\n");
+	registerDefaultCmd(WRAP_DEFAULTCOMMAND(Debugger, lingoCommandProcessor));
+	debugPrintf(PROMPT);
 	return true;
 }
 
+bool Debugger::cmdStack(int argc, const char **argv) {
+	debugPrintf("%s\n", g_director->getLingo()->formatStack().c_str());
+	debugPrintf("\n");
+	return true;
+}
+
+bool Debugger::cmdFrame(int argc, const char **argv) {
+	debugPrintf("%s\n", g_director->getLingo()->formatFrame().c_str());
+	debugPrintf("\n");
+	return true;
+}
+
+bool Debugger::cmdBacktrace(int argc, const char **argv) {
+	debugPrintf("%s\n", g_director->getLingo()->formatCallStack(g_director->getLingo()->_pc).c_str());
+	return true;
+}
+
+bool Debugger::cmdVars(int argc, const char **argv) {
+	debugPrintf("%s\n", g_director->getLingo()->formatAllVars().c_str());
+	return true;
+}
+
+bool Debugger::cmdStep(int argc, const char **argv) {
+	_step = true;
+	if (argc == 2 && atoi(argv[1]) > 0) {
+		_stepCounter = atoi(argv[1]);
+	} else {
+		_stepCounter = 1;
+	}
+	return cmdExit(0, nullptr);
+}
+
+bool Debugger::cmdNext(int argc, const char **argv) {
+	_step = true;
+	_next = true;
+	if (argc == 2 && atoi(argv[1]) > 0) {
+		_stepCounter = atoi(argv[1]);
+	} else {
+		_stepCounter = 1;
+	}
+	return cmdExit(0, nullptr);
+}
+
+bool Debugger::cmdFinish(int argc, const char **argv) {
+	_finish = true;
+	_finishCounter = 1;
+	return cmdExit(0, nullptr);
+}
+
 bool Debugger::lingoCommandProcessor(const char *inputOrig) {
-	if (!strcmp(inputOrig, "lingo off")) {
+	if (!strcmp(inputOrig, "exit")) {
```

Comment:
I am afraid that a number of Director titles could have `exit` as some meaningful keyword. This is why we had `lingo off` in the first place